lib/clientv2/ippool.go

```diff
@@ -207,3 +213,64 @@ func (_ ipPools) validateAndSetDefaults(pool *apiv2.IPPool) error {
 
 	return nil
 }
+
+// maybeEnableIPIP turns enables globally IPIP if a default setting is not already configured
+// and the pool has IPIP enabled.
+func (c ipPools) maybeEnableIPIP(ctx context.Context, pool *apiv2.IPPool) error {
+	logCxt := log.WithField("IPIPMode", pool.Spec.IPIPMode)
+	if pool.Spec.IPIPMode == apiv2.IPIPModeNever {
+		logCxt.Debug("IPIP is not enabled for this pool - no need to check global setting")
+		return nil
+	}
+
+	var err error
+	ipEnabled := true
+	for i := 0; i < maxApplyRetries; i++ {
+		logCxt.WithField("Retry", i).Debug("Checking global IPIP setting")
+		res, err := c.client.FelixConfigurations().Get(ctx, "default", options.GetOptions{})
```

Comment:
I suspect one day we'll have validation to prevent people from creating FelixConfiguration with name `foo` - it either needs to be `default` or `node.X`?

Reply:
i believe we already have that

Reply:
oh. no we don't

Reply:
in that case, yes one day.